src/app/pages/media-details/media-details.component.ts

```diff
@@ -1,28 +1,40 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, switchMap } from 'rxjs';
 import { DetailsMediaService } from 'src/app/features/media/services/details-media.service';
 import { DetailsMediaData, MediaResult } from 'src/app/features/media/types';
 
+const DEFAULT_DATA: DetailsMediaData = {type: 'movie', id: 123 }
```

Comment:
I think you do not need this defaults, your component should display expected by user media and some error if media is not found.